scopes/workspace/install/install.main.runtime.ts

```diff
@@ -442,6 +445,45 @@ export class InstallMain {
     return rootPolicy;
   }
 
+  private async addConfiguredGeneratorEnvsToWorkspacePolicy(rootPolicy: WorkspacePolicy): Promise<void> {
+    const configuredEnvs = this.generator.getConfiguredEnvs();
+    const resolvedEnvs = compact(
+      await Promise.all(
+        configuredEnvs.map(async (envIdStr) => {
+          if (this.envs.isCoreEnv(envIdStr)) {
+            return undefined;
+          }
+          const parsedId = await this.workspace.resolveComponentId(envIdStr);
+          const comps = await this.workspace.importAndGetMany(
+            [parsedId],
+            `to get the env ${parsedId.toString()} for installation`
+          );
```

Comment:
It seems to me that it would make more sense to loop over all envIdStr first and then pass the array of envIdStr to `importAndGetMany`, then do the next actions.

Reply:
you are right, but since this is usually a very small array it doesn't worth the complication of the code fmo

Reply:
@GiladShoham , if you don't want to refactor to do it one import, at least change this to run serially, not in parallel. It's risky to run import of multiple components in parallel. (we never do that). You could end up writing the same files at the same time and corrupt the objects.
The best would be of course to have one import.